HAWQ-1479 - doc ranger kerberos integration

lisakowen opened the pull request:
document ranger kerberos integration:
- remove kerberos limitation
- update ranger integ/config to include new enable-ranger-plugin.sh kerberos options, call out kerberos diffs, and link to new ranger/kerberos integration page
- new ranger/kerberos integration page identifying config procedures for 
  - RPS -> Kerberized Ranger (policy checking)
  - Ranger -> Kerberized HAWQ (resource lookup)
- call out any HA considerations with respect to ranger - not sure if i got everything

dyozie commented on markdown/ranger/ranger-kerberos.html.md.erb:
````diff
@@ -0,0 +1,209 @@
+    gpadmin@master$ . /usr/local/hawq/greenplum_path.sh
+    ```
+
+2. Identify an existing or create a new HAWQ administrative role for Ranger resource lookup. For example, to create a new administrative role:
````
Small edit:  Identify an existing **HAWQ administrative role** or create a new ...

dyozie commented on markdown/ranger/ranger-kerberos.html.md.erb:
```diff
@@ -0,0 +1,209 @@
+
+4. Start the Ranger Admin UI in a supported web browser. The default URL is \<ranger-admin-node\>:6080. 
+
+5. Locate the HAWQ service definition and press the **Edit** button. 
```
press -> click

dyozie commented on markdown/ranger/ranger-kerberos.html.md.erb:
```diff
@@ -0,0 +1,209 @@
+
+    1. Open the Ranger Admin UI. The default URL is \<ranger-admin-node\>:6080.
+    2. Navigate to **Settings > Users/Groups**.
+    3. Press the **Add New User** button.
```
Press -> Click

dyozie commented on markdown/ranger/ranger-kerberos.html.md.erb:
```diff
@@ -0,0 +1,209 @@
+
+7. Navigate to the Ranger Admin UI in a supported web browser. The default URL is \<ranger-admin-node\>:6080. 
+
+8. Locate the **HAWQ** service definition and press the **Edit** button. 
```
press -> click

dyozie commented on markdown/ranger/ranger-kerberos.html.md.erb:
```diff
@@ -0,0 +1,209 @@
+
+6. Update the applicable **Config Properties** fields:
+
+    **HAWQ User Name*** - The HAWQ Ranger lookup role you identified or created in Step 2 above.  
```
I think for consistency we should change the hyphens to colons here.  Also use sentences here:  "Enter the HAWQ Ranger lookup role ..."

dyozie commented on markdown/ranger/ranger-kerberos.html.md.erb:
```diff
@@ -0,0 +1,209 @@
+
+## Configure HAWQ for Kerberized Ranger<a id="rps2ra_kerb_cfg"></a>
+
+When Ranger Authorization is enabled for HAWQ, the Ranger Plug-in Service fetches policies from the Ranger Administration host. When Kerberos is enabled for Ranger, you must specifically configure the Ranger Plug-in Service for this communication.
```
"When Kerberos is enabled for Ranger" -> "**If** Kerberos is enabled for Ranger"

dyozie commented on markdown/ranger/ranger-kerberos.html.md.erb:
```diff
@@ -0,0 +1,209 @@
+This procedure includes:
+
+- Adding an administrative Ranger user named `postgres`
+- Updating HAWQ Ranger Plug-in Service configuration to utilize Kerberos
```
Updating **the** HAWQ Ranger ...

dyozie commented on markdown/ranger/ranger-kerberos.html.md.erb:
```diff
@@ -0,0 +1,209 @@
+    3. Press the **Add New User** button.
+    4. Fill out the required fields in the **User Detail** dialog:
+
+        **User Name*** - `postgres`  
```
Same edit as above list.

dyozie commented on markdown/ranger/ranger-kerberos.html.md.erb:
```diff
@@ -0,0 +1,209 @@
+    4. Fill out the required fields in the **User Detail** dialog:
+
+        **User Name*** - `postgres`  
+        **New Password*** - The user's password  ?? IS THIS SIGNIFICANT ??  
```
Need answer to ?? ?? here.

dyozie commented on markdown/ranger/ranger-kerberos.html.md.erb:
````diff
@@ -0,0 +1,209 @@
+        RPS_AUTH=kerberos
+        ``` 
+
+    3. `rps.properties` also includes `RPS_PRINCIPAL` and `RPS_KEYTAB` properties. If unset, the Ranger Plug-in Service uses the Kerberos credentials cache in place for the HAWQ runtime user account (typically `gpadmin`). Set these properties if the default values are not appropriate for your environment.
````
also includes **the**

If **these properties are** unset, the Ranger Plug-in Service uses the **cached Kerberos credentials that are available** for the HAWQ runtime user account (typically `gpadmin`).  <--- I *think* this edit is correct.

dyozie commented on markdown/ranger/ranger-kerberos.html.md.erb:
```diff
@@ -0,0 +1,209 @@
+
+9. Update the applicable **Config Properties** fields:
+  
+    **HAWQ Authentication Type** - Select **Kerberos** from the drop-down list  
```
Same edits as above.

interma commented on markdown/ranger/ranger-kerberos.html.md.erb:
````diff
@@ -0,0 +1,209 @@
+    $ ssh root@<kdc-server>
+    root@kdc-server$ kadmin.local -q "addprinc -pw changeme rangerlookup_hawq@REALM.DOMAIN"
+    ```
+    
````
After this step, user should validate the "rangerlookup_hawq" account:
```
kinit rangerlookup_hawq
psql -d postgres -p 5432 -U rangerlookup_hawq -h hawq-host
# check whether connect hawq successfully
```

So if there are somethings wrong, user can detect it earlier.

interma commented on markdown/ranger/ranger-kerberos.html.md.erb:
```diff
@@ -0,0 +1,209 @@
+    **HAWQ Authentication Type** : Select **Kerberos** from the drop-down list.  
+    **HAWQ Kerberos Service Name** : Enter `postgres` or your custom HAWQ service name.   
+
+10. **Save** your changes to permanently configure Kerberos authentication for Ranger policy retrieval.
```
I am a little confused: what's effects the of step7~step10？
These steps are only needed in Ranger->HAWQ (in above section), no needed in this HAWQ/RPS->Ranger section.

Right? Or do I missing something that you had communicated before?
Thanks!

lisakowen replied:
thanks for reviewing, @interma!

i had thought that:
- username and password were used when configuring ranger->kerberized hawq
- authentication type and service name were used when configuring RPS->kerberized ranger

if all four of them are only used for ranger->kerberized hawq, i will update the appropriate sections.

interma replied:
Yes, all four of them are only used for ranger->kerberized hawq (step6~9 you had wroten in "Configure Ranger for Kerberized HAWQ" section)
You should remove them in the "Configure HAWQ for Kerberized Ranger" secion.

lisakowen commented:
@interma, more on configuring the HAWQ service definition.

from rps.properties:

# RPS connect to RangerAdmin authentication type: simple or kerberos
RPS_AUTH=simple

does this RPS_AUTH correspond to the *HAWQ Authentication Type* field on the Ranger HAWQ service definition page?  if so, it would seem that you need to set that to "kerberos" when you are configuring PRS->kerberized ranger.  and that it could stay the "simple" type if you are only configuring Ranger->kerberized HAWQ.

thanks for clarifying.

interma commented:
@lisakowen about RPS_AUTH

It's **not** correspond to the HAWQ Authentication Type field on the Ranger.
So must RPS_AUTH = kerberos when ranger is kerberized, and HAWQ Authentication Type field on the Ranger can set to any value (it only depends whether HAWQ is kerberized or not).

Of course, most of user scenario are all components of cluster is kerberized, so should both of them.

Did I explain it clearly? If not, feel free to comment again!

lisakowen commented:
got it.  thanks @interma.  i have removed the hawq service definition update steps from RPS->kerberized ranger section, and moved the auth type and kerberos service name to the ranger->kerberized hawq section.
